### PR TITLE
[Merged by Bors] - Feat: add lemmas about `Set.diagonal`

### PR DESCRIPTION
**Pull request by urkud**
This is a partial forward-port of leanprover-community/mathlib#18111

### CODE REVIEW

File: Mathlib/Data/Set/Function.lean [hunk 1]
```diff
@@ -384,6 +384,9 @@ theorem mapsTo' : MapsTo f s t ↔ f '' s ⊆ t :=
   image_subset_iff.symm
 #align set.maps_to' Set.mapsTo'
 
+theorem mapsTo_prod_map_diagonal : MapsTo (Prod.map f f) (diagonal α) (diagonal β) :=
+diagonal_subset_iff.2 <| fun _ => rfl
```
Comment (fpvandoorn) [inline Mathlib/Data/Set/Function.lean:388]: don't forget to `#align`

**Comment by fpvandoorn** [comment]
Needs to be updated after new files have been ported

**Comment by fpvandoorn** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**